### ISSUE TITLE
Crashes on Android after app is resumed if running in multi-threaded mode

### ISSUE THREAD

**Issue by bdach** [ppy/osu-framework#4994]
As reported in https://github.com/ppy/osu/discussions/16512
Follow-up to #4200

Looks like the context preservation added with https://github.com/ppy/osuTK/pull/74 only works on some devices because android gonna android (which I half-expected because the android documentation [seemed to indicate that this may be the case](https://developer.android.com/reference/android/opengl/GLSurfaceView#setPreserveEGLContextOnPause(boolean))), so it looks like in the worst case scenario there is still a need for the game to be fully cycled to continue working after resume.

**Comment by bdach** [comment]
So seeing that this looks to be pretty much the number 1 issue per user/event count on sentry I went back to this one today to try and see if I can fix, but I'm having two problems:

* First and foremost, despite having an idea of how to fix it, I can actually not reproduce the context loss on any of my test devices despite numerous attempts. Apparently the loss is most common when the device goes to sleep, but it's just... not happening here. I even tried `adb` commands to force sleep to no avail.
* To know when the GL context is disposed, I'll want to hook onto [`GLSurfaceView.Renderer.onSurfaceCreated()`](https://developer.android.com/reference/kotlin/android/opengl/GLSurfaceView.Renderer#onsurfacecreated), which means I'll need to make osuTK changes - but I won't be able to use them because osuTK is hard-stuck on android after changes to compile it for net6 (https://github.com/ppy/osuTK/pull/75). So the fix ends up hard-depending on @huoyaoyuan's android on net6 series.

**Comment by bdach** [comment]
Upon further investigation reproduction requires multithreaded execution. May not be the context being lost after all, rather than it being accessed before the game is allowed to do so.

**Comment by peppy** [comment]
That `MakeCurrent()` call definitely looks suspicious (and likely unnecessary).

Does changing that alone make crashing less likely?

**Comment by bdach** [comment]
No, it only changes error 1 to error 2. It's still a 100% crash either way.

Going to poke some more but my hopes are low.

**Comment by bdach** [comment]
Had that thought too, then saw these two pages:

https://wiki.libsdl.org/Android
https://github.com/libsdl-org/SDL/blob/main/docs/README-android.md

and slowly backed away. A lot of eldritch incantations and corralling jni and what have you. I have zero clue about any of it.

**Comment by bdach** [comment]
Tentatively closing since the two pulls link above in conjunction fix all failures I could reproduce locally, so hopefully this is gone next release. If it's not, I'll reopen.

**Comment by bdach** [comment]
This came up again (https://github.com/ppy/osu/issues/20972) - reopening.

**Comment by Mou-Ikkai** [comment]
Ah, no wonder I didn't see this. It was closed.

**Comment by kit-katkittycat** [comment]
This still happens on 2025.817.0-lazer. I've analyzed `runtime.log` and found that `GameplayClockContainer` is stopped and never restarted when this glitch occurs.
[Log files](https://github.com/user-attachments/files/22095806/osulogs.zip)

**Comment by peppy** [comment]
> This still happens on 2025.817.0-lazer. I've analyzed `runtime.log` and found that `GameplayClockContainer` is stopped and never restarted when this glitch occurs. [Log files](https://github.com/user-attachments/files/22095806/osulogs.zip)

You are referring to https://github.com/ppy/osu/issues/34732, which has been resolved in tachyon and will be hotfixed in lazer soon.
